### DISCUSSION TITLE
[HELP  | BUG] Multiple OAUTH2 schemes for different user types

### DISCUSSION THREAD

**Discussion by dharik-arsath** [fastapi/fastapi#11492] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
oauth2_scheme_recruit = OAuth2PasswordBearer(tokenUrl="/auth/recruit/token")
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/auth/token")

# above I am using different token url 


async def get_current_recruit_user(
    token: Annotated[str, Depends(oauth2_scheme_recruit)],
    user_service: RecruitUserService = Depends(get_recruit_user_service),
):

async def get_current_normal_user(
        token: Annotated[str, Depends(oauth2_scheme)],
        user_service: UserService = Depends(get_user_service),
):

# depending on normal user
@auth_router.get("/protected-route")
async def protected_route(user: UserModel = Depends(get_current_normal_user)):
    print("inside protected_route")


# depending on recruit user defined above
@recruit_auth_router.get("/protected-route2")
async def protected_route(user = Depends(get_current_recruit_user)): 
    print("inside protected_route")
```

### Description

I have job based application where all recruiter based routes should get token from /auth/recruit/token and all other users should get token from /auth/token.

When having multiple oauth2 schemes, I expected it to show different url in OPENAPI docs based on how I call these functions in fastapi depends. 

but it always shows /auth/recruit/token only.

NOTE: I have two views.py file, one for normal user and other for recruiter, and I use API ROUTER in fastapi for both and combine them with fastapi include router

### Operating System

Windows

### Operating System Details

Windows 11

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.10.11

### Additional Context

![image](https://github.com/tiangolo/fastapi/assets/60572837/e3d69c0e-7cd7-4185-9766-d4fb44fe76eb)

it always points to /auth/recruit/token not matter what oauth2 scheme I depend upon

**Comment by YuriiMotov** [comment, marked as answer]
Hi!
That's because if you don't specify `scheme_name` for security scheme, FastAPI uses it's class name as a default value. And it's 'OAuth2PasswordBearer' for both of them.

https://github.com/tiangolo/fastapi/blob/1d41a7d2df5714e6598b541cac3cac5859b3ad4b/fastapi/security/oauth2.py#L376

Just specify `scheme_name` parameter:

```
oauth2_scheme_recruit = OAuth2PasswordBearer(tokenUrl="/auth/recruit/token", scheme_name="oauth2_scheme_recruit")
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/auth/token", scheme_name="oauth2_scheme")
```

There is a PR (https://github.com/tiangolo/fastapi/pull/4791) that fixes the same issue in APIKey, but it should be slightly modified and merged.

**Comment by dharik-arsath** [comment]
got it resolved, thanks for your help.
